### DISCUSSION TITLE
GitOps Incident Management

### DISCUSSION THREAD

**Discussion by bricef** [open-gitops/project#86] [General]
How do we manage an incident using GitOps? How do we apply the principles to manage an incident. What happens with changes that have to occur out-of bands? What about resolving issues with the GitOps machinery itself.

Let's have a discussion to capture some of these practices that may be of use for the community.

**Comment by lloydchang** [comment]
GitOps and Incident Management are two different subject matters.

These two subject matters shouldn't be conflated at all.

-----

GitOps isn't designed for Incident Management at all.

For example:
> [GitOps is not concerned with processes of human decision making](https://github.com/gitops-working-group/gitops-working-group/blob/51929f5d0da476fece85a0a3c04d56ed7ed93356/PRINCIPLES.md#scope)

-----

In stark contrast, Incident Management is concerned with processes of human decision making.

For example:

> **Roles**
> Incidents within a structured organization are normally dealt with by either an [incident response team](https://en.wikipedia.org/wiki/Incident_response_team) (IRT), or an [incident management team](https://en.wikipedia.org/wiki/Incident_management_team) (IMT). These are often designated beforehand or during the event, and are placed in control of the organization whilst the incident is dealt with, to restore normal functions.
> https://en.wikipedia.org/wiki/Incident_management

> **Incidents** notify the people who need to know what is happening; they also help organizations automate, manage, and track information and communications related to outages and similar problems that can impact infrastructure.
> https://support.atlassian.com/opsgenie/docs/what-is-an-incident

> **During an Incident**
> Information and processes during a major incident.
> `>` [During an Incident](https://response.pagerduty.com/during/during_an_incident/) - Information on what to do during an incident and how to constructively contribute.
> `>` [Security Incident Response](https://response.pagerduty.com/during/security_incident_response/) - Security incidents are handled differently than normal operational incidents.
> https://response.pagerduty.com/#during-an-incident

-----

Thank you @bricef @scottrigby @lotharschulz @roberthstrand

**Comment by lloydchang** [comment]
@bricef asked:
> How do we manage an incident using GitOps?

We don't.

My reason is explained in the above at https://github.com/gitops-working-group/gitops-working-group/discussions/128#discussioncomment-633177

Thanks @bricef

**Comment by scottrigby** [comment]
See https://github.com/open-gitops/documents/pull/40
